server/src/main/java/org/elasticsearch/transport/InboundDecoder.java

```diff
@@ -35,14 +35,16 @@ public class InboundDecoder implements Releasable {
     private boolean isCompressed = false;
     private boolean isClosed = false;
     private final ByteSizeValue maxHeaderSize;
+    private final boolean acceptInboundResponses;
 
     public InboundDecoder(Recycler<BytesRef> recycler) {
-        this(recycler, new ByteSizeValue(2, ByteSizeUnit.GB));
+        this(recycler, new ByteSizeValue(2, ByteSizeUnit.GB), true);
     }
 
-    public InboundDecoder(Recycler<BytesRef> recycler, ByteSizeValue maxHeaderSize) {
+    public InboundDecoder(Recycler<BytesRef> recycler, ByteSizeValue maxHeaderSize, boolean acceptInboundResponses) {
```

Comment:
I feel like this should just be `isClientChannel` or `isServerChannel`. And make the assertion symmetric (essentially reject responses on server channels and reject requests on client channels)

Also we might add a unit test for this in `InboundDecoderTests`

Reply:
Pushed 20d94ff1c07bb2a0a7a769384adc7ad6d9053c13 .
It did not feel appropriate to make all transport channel only server or only client, so there's a mix type as well. Only the RCS 2.0 channel is a server type-of channel, and all the others are mixed type. I prefer we don't extend the scope of this PR to type all channels.

Reply:
I think all channels are either client or server, never both. That said, I do agree to *not* extend the scope of this PR to all channels as Albert suggested. It can be an small follow-up and helps tightly control this PR.